src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,136 @@
+            preparedStatement.setBigDecimal(PARAMETER_INDEX_TWO, book.getPrice());
+            int executeCreate = preparedStatement.executeUpdate();
+            if (executeCreate < MIN_CREATE_OR_UPDATE_OR_REMOVE_BOOK) {
+                throw new RuntimeException(EXPECTED_CREATE + executeCreate);
+            }
+            ResultSet resultGeneratedKey = preparedStatement.getGeneratedKeys();
+            while (resultGeneratedKey.next()) {
+                int id = resultGeneratedKey.getObject(PARAMETER_INDEX_ONE, Integer.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_SAVE_A_BOOK_MSG + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(int id) {
+        try (PreparedStatement preparedStatement =
+                     getPreparedStatement(SELECT_FROM_BOOKS_WHERE_ID)) {
+            preparedStatement.setLong(PARAMETER_INDEX_ONE, id);
+            ResultSet executeQuery = getExecuteQuery(preparedStatement);
+            if (executeQuery.next()) {
+                Book book = buildBook(executeQuery);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_GET_A_BOOK_BY_ID_MSG + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        try (PreparedStatement preparedStatement = getPreparedStatement(SELECT_FROM_BOOKS)) {
+            ResultSet executeQuery = getExecuteQuery(preparedStatement);
+            while (executeQuery.next()) {
+                Book book = buildBook(executeQuery);
+                bookList.add(book);
+            }
+            return bookList;
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_GET_A_LIST_OF_BOOK_MSG, e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        try (PreparedStatement statement = getPreparedStatement(UPDATE_BOOKS_QUERY)) {
+            statement.setString(PARAMETER_INDEX_ONE, book.getTitle());
+            statement.setBigDecimal(PARAMETER_INDEX_TWO, book.getPrice());
+            statement.setInt(COLUMN_INDEX_THREE, book.getId());
+            int executeUpdate = statement.executeUpdate();
+            if (executeUpdate < MIN_CREATE_OR_UPDATE_OR_REMOVE_BOOK) {
+                throw new RuntimeException(EXPECTED_CREATE + executeUpdate);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(CAN_T_UPDATE_BOOK + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public boolean deleteById(int id) {
+        try (PreparedStatement statement = getPreparedStatement(DELETE_FROM_BOOKS_WHERE_ID)) {
+            statement.setInt(PARAMETER_INDEX_ONE, id);
```

Comment:
Ensure that `statement.setLong` is used here, as the `id` should be a `Long` type.